modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java

```diff
@@ -315,6 +320,62 @@ public void testAndPostFilter() throws Exception {
         checkBinaryLogicOperatorFilterSplitting(and);
     }
 
+    public void testSingleSortBy() throws Exception {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        SortBy[] sorts =
+                new SortBy[] {
+                    ff.sort("properties.doubleProperty", SortOrder.DESCENDING),
+                };
+
+        SimpleFeatureSource source = dataStore.getFeatureSource("ft1");
+        assertTrue(source.getQueryCapabilities().supportsSorting(sorts));
+        Query q = new Query("ft1", Filter.INCLUDE);
+        q.setSortBy(sorts);
+
+        SimpleFeatureCollection features = source.getFeatures(q);
+        SimpleFeatureIterator it = features.features();
+        List<Double> doubleValues = new ArrayList<>(3);
+        while (it.hasNext()) {
+            SimpleFeature feature = it.next();
+            doubleValues.add((Double) feature.getAttribute("properties.doubleProperty"));
+        }
+        assertEquals(doubleValues.size(), 3);
+        Double first = doubleValues.get(0);
+        Double second = doubleValues.get(1);
+        Double third = doubleValues.get(2);
+        assertTrue(first > second);
+        assertTrue(second > third);
+    }
+
+    public void testTwoSortBy() throws Exception {
```

Comment:
I would add a test or one more attribute, where one doesn't override it ignores the NULL, sorte by the next attribute and then by the next one, makes sense?

Reply:
fixed